### PR TITLE
feat: add cancel event button to event edit page

### LINKED ISSUE #268: UI: As an organizer I can cancel events and all participants who have RSVP'd will get an email notification.
Depends on #100

### PR DESCRIPTION
**Pull request by gikf**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #268

---
- Moves cancelling button to separate component.
- Adds cancelling button to event edit page: [image](https://user-images.githubusercontent.com/60067306/148791341-45cab5e9-1d1e-442a-86be-f2f44dd9a61b.png)
- Any notes are appreciated.
- Tested on local fork.

### CODE REVIEW

File: client/src/modules/dashboard/Events/components/EventCancelButton.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const data = useMemo(
+    () => ({
+      variables: { id: event.id },
+      refetchQueries: [
+        { query: EVENT, variables: { id: event.id } },
+        { query: EVENTS },
+      ],
+    }),
+    [event],
+  );
```
Comment (ojeytonwilliams) [inline client/src/modules/dashboard/Events/components/EventCancelButton.tsx:33]: I know this is a pattern we use elsewhere, but I don't think `useMemo` helps. Creating the `data` object isn't expensive.  Fetching `event` is, but that's already happened by this point.

There may be some optimisation we *can* do, but I don't think this is it.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: client/src/modules/dashboard/Events/components/EventCancelButton.tsx (hunk 1 above)
Comment (gikf) [inline client/src/modules/dashboard/Events/components/EventCancelButton.tsx:33]: Right, I followed this from elsewhere. If I understand correctly, you suggest to simply do instead:

```suggestion
  const data = {
    variables: { id: event.id },
    refetchQueries: [
      { query: EVENT, variables: { id: event.id } },
      { query: EVENTS },
    ],
  };
```

File: client/src/modules/dashboard/Events/components/EventForm.tsx [hunk 2]
```diff
@@ -236,14 +239,24 @@ const EventForm: React.FC<EventFormProps> = (props) => {
...
         </FormControl>
-        <Button
-          width="100%"
-          colorScheme="blue"
-          type="submit"
-          isDisabled={loading}
-        >
-          {submitText}
-        </Button>
+        {data && data.canceled && <Text color="red.500">Event canceled</Text>}
```
[the remark this answers is not in the document]
Comment (gikf) [reply client/src/modules/dashboard/Events/components/EventForm.tsx:242]: I appreciate all nitpicks :slightly_smiling_face:

File: client/src/modules/dashboard/Events/components/EventCancelButton.tsx (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/modules/dashboard/Events/components/EventCancelButton.tsx:33]: Yep, exactly that. Unless I'm missing something, that's essentially the same, but simpler.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**